File: examserver/src/main/java/com/exam/examserver/services/QuizService.java
package com.exam.examserver.services;

import com.exam.examserver.model.exam.Quiz;

import java.util.Set;

public interface QuizService {
    public Quiz addQuiz(Quiz quiz);

    public Quiz updateQuiz(Quiz quiz);

    public Set<Quiz> getQuizzes();

    public Quiz getQuiz(Long quizId);

    public void deleteQuiz(Long quizId);
}

File: examportfront/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http:HttpClient) { }

  public quizzes(){
    return this.http.get(`${baseUrl}/quiz/`);
  }

  //add quiz function
  public addQuiz(quiz: any){
   return this.http.post(`${baseUrl}/quiz/`,quiz);
  }

  //delete quiz function
  public deleteQuiz(qid:any){
return this.http.delete(`${baseUrl}/quiz/${qid}`);
  }

  //get the single quiz
  public getQuiz(qid: any){
    return this.http.get(`${baseUrl}/quiz/${qid}`);
  }

  //update quiz
  public updateQuiz(quiz: any){
    return this.http.put(`${baseUrl}/quiz/`,quiz);
  }
}
